common/src/legacy_memory_region.rs

```diff
@@ -28,8 +28,11 @@ pub struct LegacyFrameAllocator<I, D> {
     memory_map: I,
     current_descriptor: Option<D>,
     next_frame: PhysFrame,
+    start_frame: PhysFrame,
```

Comment:
We already use the name `start_frame` in multiple places, so we should name this field differently to avoid confusion. I think `min_frame` would be a fitting name:

```suggestion
    min_frame: PhysFrame,
```